homeassistant/components/media_player/directv.py

```diff
@@ -165,13 +167,38 @@ def update(self):
                     self._last_update = dt_util.utcnow() if not self._paused \
                         or self._last_update is None else self._last_update
                 else:
-                    self._available = False
+                    # If an error is received then only set to unavailable if
+                    # this started at least 1 minute ago.
+                    if not self._first_error_timestamp:
```

Comment:
Can we make a small helper method for the common parts of the if else block so we don't need to repeat everything twice?

Reply:
Absolutely. :-) Done.